aws/data_source_aws_default_tags.go

```diff
@@ -0,0 +1,30 @@
+package aws
+
+import "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+func dataSourceAwsDefaultTags() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsDefaultTagsRead,
+
+		Schema: map[string]*schema.Schema{
+			"tags": tagsSchemaComputed(),
+		},
+	}
+}
+
+func dataSourceAwsDefaultTagsRead(d *schema.ResourceData, meta interface{}) error {
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	d.SetId(meta.(*AWSClient).partition)
+
+	if defaultTagsConfig == nil || defaultTagsConfig.Tags == nil {
+		return nil
+	}
+
+	if err := d.Set("tags", defaultTagsConfig.Tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return err
+	}
```

Comment:
i'm not 100% on how we set empty values, but we could also always set the "tags" field if we do something like
```go
tags := defaultTagsConfig.GetTags()

...

if len(tags) > 0 {
    tags = tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig)
}

if err := d.Set("tags", tags.Map()); err != nil {		
    return fmt.Errorf("error setting tags": %w, err)
}
```

Reply:
only trouble with that is semgrep doesn't like it - ignore must accompany `d.Set("tags"...)`